src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditProceedingsAndCostsSectionController.java

```diff
@@ -755,4 +822,379 @@ public String proceedingsActionConfirmPost(
+      final Model model,
+      final ApplicationDetail application,
+      final ProceedingFlowFormData proceedingFlow) {
+
+    final ScopeLimitationDetail criteria = new ScopeLimitationDetail()
+        .categoryOfLaw(application.getCategoryOfLaw().getId())
+        .matterType(proceedingFlow.getMatterTypeDetails().getMatterType())
+        .proceedingCode(proceedingFlow.getProceedingDetails().getProceedingType())
+        .levelOfService(proceedingFlow.getFurtherDetails().getLevelOfService());
+
+    if (EMERGENCY_APPLICATION_TYPE_CODES.contains(application.getApplicationType().getId())) {
+      criteria.emergency(true);
+    }
+
+    final List<ScopeLimitationDetail> scopeLimitationTypes = Optional.ofNullable(
+            lookupService.getScopeLimitationDetails(criteria).block())
+        .map(result ->
+            result.getContent() != null
+                ? result.getContent().stream()
+                .sorted(Comparator.comparing(ScopeLimitationDetail::getDescription))
+                .collect(Collectors.toList())
+                : null)
+        .orElse(Collections.emptyList());
+
+    model.addAttribute("scopeLimitationTypes", scopeLimitationTypes);
+
+  }
+
+  /**
+   * Populates the model with detailed scope limitation information for the current
+   * application and proceeding flow.
+   *
+   * <p>Builds criteria based on application details, proceeding flow, and selected scope
+   * limitations to retrieve a specific scope limitation detail. Handles emergency
+   * application criteria separately. Maps the retrieved detail to a scope limitation
+   * object and updates the model for view rendering.</p>
+   *
+   * @param model the {@link Model} object for passing attributes to the view.
+   * @param application the application details, used to set criteria.
+   * @param proceedingFlow the proceeding flow data, used to set criteria.
+   * @param scopeLimitationFlow the selected scope limitation data.
+   */
+  private void populateScopeLimitationDetails(
+      final Model model,
+      final ApplicationDetail application,
+      final ProceedingFlowFormData proceedingFlow,
+      final ScopeLimitationFlowFormData scopeLimitationFlow) {
+
+    final ScopeLimitationDetail criteria = new ScopeLimitationDetail()
+        .categoryOfLaw(application.getCategoryOfLaw().getId())
+        .matterType(proceedingFlow.getMatterTypeDetails().getMatterType())
+        .proceedingCode(proceedingFlow.getProceedingDetails().getProceedingType())
+        .levelOfService(proceedingFlow.getFurtherDetails().getLevelOfService())
+        .scopeLimitations(scopeLimitationFlow.getScopeLimitationDetails().getScopeLimitation());
+
+    if (EMERGENCY_APPLICATION_TYPE_CODES.contains(application.getApplicationType().getId())) {
+      criteria.emergency(true);
+    }
+
+    final ScopeLimitationDetail scopeLimitationDetail = Optional.ofNullable(
+        lookupService.getScopeLimitationDetails(criteria))
+        .map(Mono::block)
```

Comment:
Could do a .block() to keep it the same as the above method

Reply:
amended